### ISSUE TITLE
[Feature Request] exclude look-alike characters in password generator

### ISSUE THREAD

**Issue by r0bj** [mstarke/MacPass#622]
In the keepassx (https://www.keepassx.org/) there is very nice feature which is called "Exclude look-alike characters" in password generator.

It allows to avoid similar characters like:
- "I" (ASCII 108) and "I" (ASCII 73)
- "O" (ASCII 79) and "0" (ASCII 48)
etc.

Depending on used fonts it's sometimes difficult to distinguish such characters which can lead to errors.

**Comment by mstarke** [comment]
Thinking about this, the solution 1Password chose comes to mind. They use color coding to help you understand the Password better:
![screen shot 2017-09-06 at 19 50 35](https://user-images.githubusercontent.com/58428/30126605-ae4d6036-933c-11e7-934d-cbc6640c0dd7.png)

**Comment by mstarke** [comment]
Color-coding works but integration into the password fields is a bit harder than I anticipated. But it's making progress.

**Comment by tflo** [comment]
Yes, an option to exclude look-alike characters would be handy. I often catch myself recreating a password when it contains to many similar characters, especially when I know there might be chances that I will have to enter that password by hand.

**Comment by mwright-oss** [comment]
The colour coding would be a beautiful solution, but it's a fair bit of work to implement and it would fail when the end user can only display the password in a monochrome typeface which doesn't properly distinguish between letters.

A simple and elegant solution would be to add an option to say simply:

Exclude Characters:  o 0 O i l | 1 L

I would go as far as saying that this should be the default - how many times have we been unable to copy/paste the password and had to squint at it to try and try to guess what the letters are. Loss of a couple of bits of key space is a small price to pay. 

Danke!

**Comment by mstarke** [comment]
I still wasn't able to fully implement this but here's a dev-peek. Colors are not ideal :)

![screen shot 2018-06-21 at 10 02 49](https://user-images.githubusercontent.com/58428/41705797-64297810-753a-11e8-9a4f-bfc82286fb85.png)

**Comment by mstarke** [comment]
The font is monospaced already. But only on the password and username filed. The custom fields use normal fonts. It might be usefull to switch all fields to a monospaced font.

**Comment by mwright-oss** [comment]
Looks good. Try using #006600 for the green so the contrast is better when it's displayed on a white background. 

Mono fonts are a good choice, my old Mac favourite from many years ago was Monaco. DejaVu Sans Mono is very good too.

It would be good to see the feature to Exclude Characters: o 0 O i l | 1 L  - we can't do much about existing passwords but it would be good to avoid generating random passwords with potentially ambiguous characters because not everyone is fortunate/smart enough to have clear typefaces to display it.
